src/domain/opendatafabric/src/identity/dataset_identity.rs

```diff
@@ -305,13 +317,19 @@ newtype_str!(
     AccountNameSerdeVisitor
 );
 
+impl AccountName {
+    pub fn lowercase_eq(&self, other: &Self) -> bool {
+        self.to_lowercase() == other.to_lowercase()
+    }
+}
+
 ////////////////////////////////////////////////////////////////////////////////
 
 newtype_str!(RepoName, Grammar::match_repo_name, RepoNameSerdeVisitor);
 
 ////////////////////////////////////////////////////////////////////////////////
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Debug, Clone, Eq, PartialOrd, Ord)]
```

Comment:
If we are customizing `PartialEq`, shouldn't we also customize `PartialOrd` ? I assume it might be useful for correct ignore-case sorting.

Reply:
Due to the lastest changes, I moved the custom PartialEq  implementation to `newtype_istr` level